thunder/dynamo/benchmark_utils.py

```diff
@@ -224,6 +227,33 @@ def cleanup(self):
             pass
 
 
+class TimerWithCUDAMemoryUsage:
+    """
+    A timer wrapper that tracks CUDA memory usage alongside timing measurements.
```

Comment:
A link to https://pytorch.org/docs/stable/notes/cuda.html#memory-management might help explain how PyTorch measures memory usage